### PR TITLE
[DEV-2435] adapt swap aml rules

### PR DESCRIPTION
**Pull request by Yannick1712**
#### Pre-Release

- [ ] Check migrations
  - No database related infos (sqldb-xxx)
  - Impact on GS (new/removed columns)
- [ ] Check for linter errors (in PR)
- [ ] Test basic user operations (on [DFX services](https://dev.services.dfx.swiss))
  - Login/logout
  - Buy/sell payment request
  - KYC page

#### Post-Release

- Test basic user operations
- Monitor application insights log

### CODE REVIEW

File: src/subdomains/core/buy-crypto/routes/swap/dto/get-crypto-payment-info.dto.ts
Comment (davidleomay) [inline src/subdomains/core/buy-crypto/routes/swap/dto/get-crypto-payment-info.dto.ts:1]: Rename?

File: src/subdomains/core/buy-crypto/routes/swap/dto/get-crypto-quote.dto.ts
Comment (davidleomay) [inline src/subdomains/core/buy-crypto/routes/swap/dto/get-crypto-quote.dto.ts:1]: Rename?

File: src/subdomains/core/buy-crypto/routes/swap/swap.controller.ts [hunk 1]
```diff
@@ -48,43 +48,43 @@ export class CryptoRouteController {
...
   }
 
-  // TODO: enable quote + info endpoints after refactoring (move to convert)
+  // // TODO: enable quote + info endpoints after refactoring (move to convert)
   @ApiExcludeEndpoint()
   @Put('/quote')
-  @ApiOkResponse({ type: CryptoQuoteDto })
-  async getCryptoQuote(@Body() dto: GetCryptoQuoteDto): Promise<CryptoQuoteDto> {
+  @ApiOkResponse({ type: SwapQuoteDto })
+  async getCryptoQuote(@Body() dto: GetCryptoQuoteDto): Promise<SwapQuoteDto> {
```
Comment (davidleomay) [inline src/subdomains/core/buy-crypto/routes/swap/swap.controller.ts:80]: Remove this comment and the `ApiExcludeEndpoint`

File: src/subdomains/core/buy-crypto/routes/swap/swap.controller.ts [hunk 2]
```diff
@@ -130,12 +130,12 @@ export class CryptoRouteController {
   @Put('/paymentInfos')
   @ApiBearerAuth()
   @UseGuards(AuthGuard(), new RoleGuard(UserRole.USER), IpGuard)
-  @ApiOkResponse({ type: CryptoPaymentInfoDto })
+  @ApiOkResponse({ type: SwapPaymentInfoDto })
   async createCryptoWithPaymentInfo(
```
Comment (davidleomay) [inline src/subdomains/core/buy-crypto/routes/swap/swap.controller.ts:134]: Remove `ApiExcludeEndpoint` above

File: src/subdomains/core/buy-crypto/routes/swap/swap.service.ts
Comment (davidleomay) [inline src/subdomains/core/buy-crypto/routes/swap/swap.service.ts:1]: Rename methods?

File: src/subdomains/core/aml/aml-helper.service.ts [hunk 3]
```diff
@@ -66,8 +66,6 @@ export class AmlHelperService {
       // crypto input
       if (entity.cryptoInput.amlCheck !== CheckStatus.PASS) errors.push(AmlError.INPUT_AML_CHECK_FAILED);
       if (!entity.cryptoInput.isConfirmed) errors.push(AmlError.INPUT_NOT_CONFIRMED);
-      if (entity instanceof BuyCrypto && !entity.userData.cryptoCryptoAllowed)
-        errors.push(AmlError.CRYPTO_CRYPTO_NOT_ALLOWED);
```
Comment (davidleomay) [inline src/subdomains/core/aml/aml-helper.service.ts:70]: Remove `cryptoCryptoAllowed` on UserData? Or maybe in a second step?

File: src/subdomains/supporting/payment/services/transaction-helper.ts [hunk 4]
```diff
@@ -437,10 +437,21 @@ export class TransactionHelper implements OnModuleInit {
...
+    }
+
     if (paymentMethodIn === FiatPaymentMethod.INSTANT && user && !user.userData.olkypayAllowed)
       return QuoteError.KYC_REQUIRED_INSTANT;
     if (
-      ((isFiat(to) && to.name !== 'CHF') || paymentMethodIn === FiatPaymentMethod.CARD) &&
+      ((isFiat(to) && to.name !== 'CHF') || paymentMethodIn === FiatPaymentMethod.CARD || isSwapTx) &&
       user &&
       !user.userData.hasBankTxVerification &&
       txAmountChf > Config.tradingLimits.dailyDefault
```
Comment (davidleomay) [inline src/subdomains/supporting/payment/services/transaction-helper.ts:457]: Move the amount checks up (above isSwapTx) and remove the `QuoteError.DEPOSIT_LIMIT_REACHED`. `(txAmountChf > Config.tradingLimits.dailyDefault && user?.userData.kycLevel < KycLevel.LEVEL_50)` can also be removed.

File: src/subdomains/core/aml/aml-helper.service.ts (hunk 3 above)
Comment (Yannick1712) [reply src/subdomains/core/aml/aml-helper.service.ts:70]: I would do it in another step

Comment (Yannick1712) [reply src/subdomains/core/aml/aml-helper.service.ts:70]: I made a ticket for me for it 👍

File: src/subdomains/supporting/payment/services/transaction-helper.ts [hunk 5]
```diff
@@ -437,18 +437,24 @@ export class TransactionHelper implements OnModuleInit {
         return QuoteError.KYC_REQUIRED;
     }
 
+    // amount checks
+    if (sourceAmount < txSourceMinVolume) return QuoteError.AMOUNT_TOO_LOW;
+    if (txAmountChf > maxVolumeChf) return QuoteError.AMOUNT_TOO_HIGH;
```
Comment (davidleomay) [inline src/subdomains/supporting/payment/services/transaction-helper.ts:442]: Move down

**Review by davidleomay — APPROVED** [review]

**Pull request merged**